src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -721,7 +721,7 @@ ssize_t sockinfo_tcp::tx(const tx_call_t call_type, const struct iovec* p_iov, c
 
 	if (unlikely(is_dummy) && !check_dummy_send_conditions(flags, p_iov, sz_iov)) {
 		unlock_tcp_con();
-		errno = EIO;
+		errno = EOPNOTSUPP;
```

Comment:
@liranoz12 all write operations as send/sendto condifer `EOPNOTSUPP` as special errno for situations when passed flags do not meet expected 
request.
`
The socket argument is associated with a socket that does not support one or more of the values set in flags.
`
At the same time 
`
EIO
An I/O error occurred while reading from or writing to the file system.`